Data Race when creating a Cluster with seed topics

We're getting some data race errors when creating a cluster with seed topics on Github Actions.
Looks like the race is between handling requests and creating the seed topics.

(you can find the build here: https://github.com/steadybit/extension-kafka/actions/runs/12052475922/job/33605865400)

```
==================
WARNING: DATA RACE
Read at 0x00c00014a798 by goroutine 2069:
  github.com/twmb/franz-go/pkg/kfake.(*tps[go.shape.dd00dc7b83e51d21898f331f24c492b0a430c1f9c6c0591a903ca815b53ebec5]).gett()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/topic_partition.go:21 +0xc66
  github.com/twmb/franz-go/pkg/kfake.(*Cluster).handleMetadata()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/03_metadata.go:95 +0xc18
  github.com/twmb/franz-go/pkg/kfake.(*Cluster).run()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/cluster.go:361 +0x877
  github.com/twmb/franz-go/pkg/kfake.NewCluster.gowrap2()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/cluster.go:189 +0x33

Previous write at 0x00c00014a798 by goroutine 2064:
  github.com/twmb/franz-go/pkg/kfake.(*tps[go.shape.dd00dc7b83e51d21898f331f24c492b0a430c1f9c6c0591a903ca815b53ebec5]).mkt()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/topic_partition.go:27 +0x93
  github.com/twmb/franz-go/pkg/kfake.(*tps[go.shape.dd00dc7b83e51d21898f331f24c492b0a430c1f9c6c0591a903ca815b53ebec5]).mkp()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/topic_partition.go:38 +0x48
  github.com/twmb/franz-go/pkg/kfake.(*data).mkt()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/data.go:99 +0x555
  github.com/twmb/franz-go/pkg/kfake.NewCluster()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/cluster.go:202 +0x1ce4
  github.com/steadybit/extension-kafka/extkafka.TestNewHTTPCheckActionPeriodically_All_Failure()
      /home/runner/work/extension-kafka/extension-kafka/extkafka/producePeriodically_test.go:183 +0x2f9
  testing.tRunner()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1743 +0x44

Goroutine 2069 (running) created at:
  github.com/twmb/franz-go/pkg/kfake.NewCluster()
      /home/runner/go/pkg/mod/github.com/twmb/franz-go/pkg/kfake@v0.0.0-20241010233600-b66ceb79e585/cluster.go:189 +0x1a8a
  github.com/steadybit/extension-kafka/extkafka.TestNewHTTPCheckActionPeriodically_All_Failure()
      /home/runner/work/extension-kafka/extension-kafka/extkafka/producePeriodically_test.go:183 +0x2f9
  testing.tRunner()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1743 +0x44

Goroutine 2064 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.23.3/x64/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:85 +0x164
make: *** [Makefile:40: audit] Error 1
==================
```

--- a/pkg/kfake/cluster.go
+++ b/pkg/kfake/cluster.go
@@ -183,10 +183,9 @@ func NewCluster(opts ...Opt) (*Cluster, error) {
 			bsIdx: len(c.bs),
 		}
 		c.bs = append(c.bs, b)
-		go b.listen()
+		defer func() { go b.listen() }()
 	}
 	c.controller = c.bs[len(c.bs)-1]
-	go c.run()
 
 	seedTopics := make(map[string]int32)
 	for _, sts := range cfg.seedTopics {
@@ -201,6 +200,9 @@ func NewCluster(opts ...Opt) (*Cluster, error) {
 	for t, p := range seedTopics {
 		c.data.mkt(t, int(p), -1, nil)
 	}
+
+	go c.run()
+
 	return c, nil
 }
 

